### DISCUSSION TITLE
Using tflint with Docker

### DISCUSSION THREAD

**Discussion by vibaiher-qatium** [terraform-linters/tflint#1908] [Q&A] [answered]
Hello everyone,

I used tflint previously with docker image, but recently I'm not able to use it, and I don't understand what is happening. I'm opening a discussion instead of a bug, because my odds are that there is wrong in my setup.

I'm using OSX, with Docker Desktop, and when I run docker pull or docker run, it says "access denied":

```
$ docker pull ghcr.io/terraform-linters/tflint-bundle:v0.48.0.0
Error response from daemon: Head "https://ghcr.io/v2/terraform-linters/tflint-bundle/manifests/v0.48.0.0": denied: denied

$ docker run --rm -v $(pwd):/data -t ghcr.io/terraform-linters/tflint
Unable to find image 'ghcr.io/terraform-linters/tflint:latest' locally
docker: Error response from daemon: Head "https://ghcr.io/v2/terraform-linters/tflint/manifests/latest": denied: denied.
See 'docker run --help'.
```

**Comment by bendrucker** [comment, marked as answer]
Did you use `docker login` at any point for `ghcr.io`? This request should work without authentication, the error suggests that credentials are being sent but they're invalid. 

To remove them, `docker logout ghcr.io`.

**Reply by bendrucker** [reply]
In general it's a good idea to `login` with a GitHub token with minimal (package) privileges. But if those credentials are revoked or lack the right scopes you'll get denied errors back from GitHub. 

GitHub gives you greater API rate limits when authenticated, versus rate limiting you by IP at a lower level when anonymous. IIRC that didn't apply to GH's registry back when Docker started rate limiting anonymous requests to DockerHub. But it might in the future.

**Reply by vibaiher-qatium** [reply]
Thanks for your help. I tried to logout, but that does not solve the issue:

```
$ docker logout
Removing login credentials for https://index.docker.io/v1/
$ docker pull ghcr.io/terraform-linters/tflint
Using default tag: latest
Error response from daemon: Head "https://ghcr.io/v2/terraform-linters/tflint/manifests/latest": denied: denied
```

I logged in with my personal user, and didn't worked:

```
$ docker login
Login with your Docker ID to push and pull images from Docker Hub. If you don't have a Docker ID, head over to https://hub.docker.com to create one.
Username: vibaiher
Password:
Login Succeeded

Logging in with your password grants your terminal complete access to your account.
For better security, log in with a limited-privilege personal access token. Learn more at https://docs.docker.com/go/access-tokens/
$ docker pull ghcr.io/terraform-linters/tflint
Using default tag: latest
Error response from daemon: Head "https://ghcr.io/v2/terraform-linters/tflint/manifests/latest": denied: denied
```

**Reply by vibaiher-qatium** [reply]
My docker config does not contain nothing about ghcr.io

```
$ cat .docker/config.json
{
	"credsStore": "desktop",
	"currentContext": "desktop-linux",
	"plugins": {
		"-x-cli-hints": {
			"enabled": "true"
		}
	}
}
```

**Reply by bendrucker** [reply]
It would have to be `docker logout ghcr.io`. But it should show up under `auths` in your Docker config at `~/.docker/config.json`. No idea what's up here. If you Google around you find various people with bad credentials who just needed to either log out or provide valid ones.

**Reply by vibaiher-qatium** [reply]
Thank you, once I solve it I post it the solution.

**Reply by vibaiher-qatium** [reply]
I solved it running this command again: `docker logout ghcr.io`

Not sure where were those credentials stored...
